ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -529,15 +530,26 @@ newBackingStore ::
   -> SomeHasFS m
   -> LedgerTables l ValuesMK
   -> m (LedgerBackingStore m l)
-newBackingStore tracer bss someHasFS tables = do
-  store <- case bss of
+newBackingStore tracer bss someHasFS tables =
+  case bss of
     LMDBBackingStore limits ->
-      LMDB.newLMDBBackingStore
+      LedgerBackingStore <$> LMDB.newLMDBBackingStore
       tracer
       limits
       someHasFS
       (LMDB.LIInitialiseFromMemory Origin tables)
-    InMemoryBackingStore -> BackingStore.newTVarBackingStore
+    InMemoryBackingStore -> createTVarBackingStore Origin tables
+
+createTVarBackingStore ::
```

Comment:
I think extracting this is not needed because the mempool doesn't create a backing store anymore